integration-tests/bats/filter-branch.bats

```diff
@@ -531,3 +532,124 @@ SQL
     [[ "$output" =~ "3,3," ]] || false
 }
 
+@test "filter-branch: --apply-to-uncommitted applies changes to unstaged tables on current branch" {
+    dolt sql -q "insert into test values (3, 3)"
+
+    run dolt sql -r csv -q "select * from dolt_status"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "table_name,staged,status" ]] || false
+    [[ "$output" =~ "test,false,modified" ]] || false
+
+    run dolt filter-branch --apply-to-uncommitted -q "alter table test add column filter int"
+    [ "$status" -eq 0 ]
+
+    run dolt sql -r csv -q "select * from dolt_status"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "table_name,staged,status" ]] || false
+    [[ "$output" =~ "test,false,modified" ]] || false
+
+    run dolt sql -r csv -q "select * from test as of 'HEAD'"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "pk,c0,filter" ]] || false
+    [[ "$output" =~ "0,0," ]] || false
+    [[ "$output" =~ "1,1," ]] || false
+    [[ "$output" =~ "2,2," ]] || false
+    [[ ! "$output" =~ "3,3," ]] || false
+}
+
+@test "filter-branch: --apply-to-uncommitted applies changes to staged tables on current branch" {
+    dolt sql -q "insert into test values (3, 3)"
+    dolt add .
+
+    run dolt sql -r csv -q "select * from dolt_status"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "table_name,staged,status" ]] || false
+    [[ "$output" =~ "test,true,modified" ]] || false
+
+    run dolt filter-branch --apply-to-uncommitted -q "alter table test add column filter int"
+    [ "$status" -eq 0 ]
+
+    run dolt sql -r csv -q "select * from dolt_status"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "table_name,staged,status" ]] || false
+    [[ "$output" =~ "test,true,modified" ]] || false
+
+    run dolt sql -r csv -q "select * from test as of 'HEAD'"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "pk,c0,filter" ]] || false
+    [[ "$output" =~ "0,0," ]] || false
+    [[ "$output" =~ "1,1," ]] || false
+    [[ "$output" =~ "2,2," ]] || false
+    [[ ! "$output" =~ "3,3," ]] || false
+}
+
+@test "filter-branch: --apply-to-uncommitted applies changes to unstaged tables on other branch" {
+    dolt sql << SQL
+call dolt_checkout('-b', 'other');
+insert into test values (3, 3);
+SQL
+
+    # no changes on main branch
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+
+    run dolt filter-branch --apply-to-uncommitted --branches -q "alter table test add column filter int"
+    [ "$status" -eq 0 ]
+
+    # still no changes on main branch
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+
+
+    dolt checkout other
```

Comment:
This doesn't do what you want. You need to switch branches in the SQL context, then see if the current working set has the new column and the new row. Changing branches with the CLI overwrites any working set changes on that branch in some circumstances (not sure if that's happening here)